Sources/Sentry/SentryUIDeviceWrapper.m

```diff
@@ -0,0 +1,67 @@
+#import "SentryUIDeviceWrapper.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface
+SentryUIDeviceWrapper ()
+@property (nonatomic) BOOL cleanupDeviceOrientationNotifications;
+@property (nonatomic) BOOL cleanupBatteryMonitoring;
+@end
+
+@implementation SentryUIDeviceWrapper
+
+#if TARGET_OS_IOS
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        // Needed to read the device orientation on demand
+        if (!UIDevice.currentDevice.isGeneratingDeviceOrientationNotifications) {
+            self.cleanupDeviceOrientationNotifications = YES;
+            [UIDevice.currentDevice beginGeneratingDeviceOrientationNotifications];
+        }
+
+        // Needed so we can read the battery level
+        if (!UIDevice.currentDevice.isBatteryMonitoringEnabled) {
+            self.cleanupBatteryMonitoring = YES;
+            UIDevice.currentDevice.batteryMonitoringEnabled = YES;
+        }
+    }
+    return self;
+}
+
+- (void)dealloc
+{
+    if (self.cleanupDeviceOrientationNotifications) {
+        [UIDevice.currentDevice endGeneratingDeviceOrientationNotifications];
+    }
+    if (self.cleanupBatteryMonitoring) {
+        UIDevice.currentDevice.batteryMonitoringEnabled = NO;
+    }
```

Comment:
h: What if the SDK inits before the user enables both orientation notification and battery monitoring. So I guess this would disabled it even if the users need it. 
I guess, in the majority of cases, this class will not be deallocated during app usage. But if you have a solution for this it would be better.

Reply:
I'm not sure I am following your thinking here. If the SentryClient inits, then we enable monitoring. The user doesn't do this. Then when the SentryClient deallocs, we clean up after ourselves.

Reply:
Ah I see what you mean. If we enable it, the user ALSO enables it, then we dealloc and stop the monitoring and also disable it for the user. Well yeah, not sure how we would work around that? And also not sure if this would happen in real life?

Reply:
If this is not easily achievable, Im fine with the way it is right now. We just need to update the docs to point this out I guess.